Feat/refactor meeting muse

thisisarjun opened the pull request:
Major refactor and code fixes

thisisarjun commented:
@ismailmmd 
I have removed several unused methods( check this [commit](https://github.com/thisisarjun/meetingmuse/pull/22/commits/4d23fd02e375a4c532aa58dc2e32a0abb1ad994e)), some of them might be useful but lets only add what we require right now, this would make debugging easier

ismailmmd commented:
@thisisarjun Actions are failing please check
```shell
poetry run flake8 src/ tests/
The virtual environment found in /home/runner/work/meetingmuse/meetingmuse/.venv seems to be broken.
Recreating virtualenv meetingmuse in /home/runner/work/meetingmuse/meetingmuse/.venv
Command not found: flake8
make: *** [Makefile:95: lint] Error 1
```

ismailmmd commented:
> @ismailmmd I have removed several unused methods( check this [commit](https://github.com/thisisarjun/meetingmuse/pull/22/commits/4d23fd02e375a4c532aa58dc2e32a0abb1ad994e)), some of them might be useful but lets only add what we require right now, this would make debugging easier

I totally agree with removing the admin APIs, in fact I had #28 for this 🫂 . IMO we could have kept the `/health` API. This would be useful for deployment and status checks - https://render.com/docs/health-checks

thisisarjun commented:
> > @ismailmmd I have removed several unused methods( check this [commit](https://github.com/thisisarjun/meetingmuse/pull/22/commits/4d23fd02e375a4c532aa58dc2e32a0abb1ad994e)), some of them might be useful but lets only add what we require right now, this would make debugging easier
> 
> I totally agree with removing the admin APIs, in fact I had #28 for this 🫂 . IMO we could have kept the `/health` API. This would be useful for deployment and status checks - https://render.com/docs/health-checks

added health api back 🔆

ismailmmd commented on src/server/services/connection_manager.py:
```diff
@@ -139,17 +141,16 @@ async def send_system_message(
-                "timestamp": datetime.now().isoformat(),
-            }
+            # TODO: system message has to be pydantic model
+            system_message = SystemMessage(
```
We are returning different types of messages in the same connection, So how would client know which message they recieve or what keys it has. So IMO `type` would be needed here. WDYT?

thisisarjun replied:
type is still there, when you make SystemMessage, the default value is type="system"

ismailmmd commented on src/server/api/app.py:
```diff
@@ -94,14 +131,6 @@ def create_app() -> FastAPI:
-            {
-                "name": "admin",
-                "description": "Administrative operations for managing connections, broadcasting messages, and retrieving system statistics.",
-            },
             {
                 "name": "websocket",
```
```suggestion
        tags_metadata=[
              {
                "name": "health",
                "description": "Health check and system monitoring endpoints. Use these for service discovery, load balancer health checks, and system observability.",
            },
            {
                "name": "websocket",
```

ismailmmd commented on src/common/utils/utils.py:
```diff
@@ -20,15 +22,15 @@ def get_last_message(
         for message in reversed(state.messages):
-            if input_type == "human" and isinstance(message, HumanMessage):
+            if input_type == "human" and message.type == "human":
                 # Handle both string and complex content types
                 content = message.content
                 if isinstance(content, str):
                     last_message = content
                 else:
                     last_message = str(content)
                 break
-            if input_type == "ai" and isinstance(message, AIMessage):
+            if input_type == "ai" and message.type == "ai":
```
Can we create constants for `human` and `ai`. Since its used in multiple places

ismailmmd commented on src/common/logger/logger.py:
Why not we make use of existing libraries like
- https://github.com/Delgan/loguru
- https://pypi.org/project/coloredlogs/
- https://pypi.org/project/colorlog/

thisisarjun replied:
can do, this was older code, maynot be important now.

ismailmmd commented on src/server/services/websocket_connection_service.py:
```diff
@@ -105,17 +107,19 @@ async def handle_websocket_connection(
+            try:
+                user_message = MessageProtocol.parse_user_message(message_text)
+            except Exception as e:
+                self.logger.error(f"Error parsing user message: {str(e)}")
+                self.logger.warning(f"Invalid message format from {client_id}")
```
```suggestion
                self.logger.warning(f"Invalid message format from {client_id}")
```
